invokeai/app/invocations/latent.py

```diff
@@ -1228,3 +1229,54 @@ def invoke(self, context: InvocationContext) -> LatentsOutput:
         context.services.latents.save(name, cropped_latents)
 
         return build_latents_output(latents_name=name, latents=cropped_latents)
+
+
+@invocation_output("ideal_size_output")
+class IdealSizeOutput(BaseInvocationOutput):
+    """Base class for invocations that output an image"""
+
+    width: int = OutputField(description="The ideal width of the image in pixels")
+    height: int = OutputField(description="The ideal height of the image in pixels")
+
+
+@invocation(
+    "ideal_size",
+    title="Ideal Size",
+    tags=["latents", "math", "ideal_size"],
+    version="1.0.2",
+)
+class IdealSizeInvocation(BaseInvocation):
+    """Calculates the ideal size for generation to avoid duplication"""
+
+    width: int = InputField(default=1024, description="Target width")
+    height: int = InputField(default=576, description="Target height")
+    unet: UNetField = InputField(default=None, description="UNet submodel")
+    multiplier: float = InputField(default=1.0, description="Dimensional multiplier")
```

Comment:
Can these descriptions be a bit more detailed? "Dimensional multiplier" in particular does not tell me anything without reading the code (and even then the intention isn't clear)

Reply:
Just updated them, hope they work.